File: Phind-CodeLlama-34B-v2/context-template-full/acc_memcpy_device.c
#include "acc_testsuite.h"
#include <stdio.h>
#include <stdlib.h>
#include <openacc.h>

#ifndef T1
/*T1:acc memcpy device,V:2.0-2.7*/
int test1(){
    int err = 0;
    srand(SEED);

    int *a, *b, *c;
    int size = 1024;
    a = (int *)malloc(size * sizeof(int));
    b = (int *)malloc(size * sizeof(int));
    c = (int *)acc_malloc(size * sizeof(int));

    for (int x = 0; x < size; ++x) {
        a[x] = rand() % size;
        b[x] = 0;
    }

    acc_memcpy_device(c, a, size * sizeof(int));
    acc_memcpy_device(b, c, size * sizeof(int));

    for (int x = 0; x < size; ++x) {
        if (a[x] != b[x]) {
            err = 1;
            break;
        }
    }

    free(a);
    free(b);
    acc_free(c);

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: Phind-CodeLlama-34B-v2/stage2/acc_memcpy_device.f90
#ifndef T1
!T1:acc memcpy device,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        INTEGER(KIND=8), PARAMETER :: N = 1024
        REAL(KIND=8), DIMENSION(N) :: a, b, c
        TYPE(C_PTR) :: a_dev, b_dev

        a = 1.0
        b = 2.0

        ! Allocate device memory
        a_dev = acc_malloc(N * 8)
        b_dev = acc_malloc(N * 8)

        ! Copy data to device
        CALL acc_memcpy_device(a_dev, C_LOC(a), N * 8)
        CALL acc_memcpy_device(b_dev, C_LOC(b), N * 8)

        ! Perform element-wise addition on device
        CALL acc_map_data(c_dev, N * 8)
        CALL acc_map_data(a_dev, N * 8)
        CALL acc_map_data(b_dev, N * 8)
        !$acc parallel loop
        DO i = 1, N
          c(i) = a(i) + b(i)
        END DO
        !$acc end parallel loop
        CALL acc_unmap_data(c_dev)
        CALL acc_unmap_data(a_dev)
        CALL acc_unmap_data(b_dev)

        ! Copy result back to host
        CALL acc_memcpy_device(C_LOC(c), c_dev, N * 8)

        ! Check result
        DO i = 1, N
          IF (c(i) .ne. 3.0) THEN
            errors = errors + 1
          END IF
        END DO

        ! Free device memory
        CALL acc_free(a_dev)
        CALL acc_free(b_dev)

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END
#endif

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
#ifndef T1
        LOGICAL :: test1
#endif
        failed = .FALSE.
        failcode = 0
#ifndef T1
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
#endif
        CALL EXIT (failcode)
      END PROGRAM